README.md

```diff
@@ -1,16 +1,226 @@
+default classes (`String`, `Integer` and many others!) can do a lot of work for
+you so that you can focus on your dreams instead of writing "external" methods
+to count `String` length.
+
+## Define Attribute / Property
+
+An attribute on an instance is a fixed bit of information that the instance can
+provide you. In Ruby, all instances have an attribute called `.class`. `String`s
+in Ruby also have the attribute or property called `length`.
+
+```ruby
+2.3.3 :003 > "hi".length
+#=> 2
+2.3.3 :004 > x = "hi"
+#=> "hi"
+2.3.3 :005 > x.length
+#=> 2
+```
+
+`Integer`s _do not_ have the `length` attribute.
+
+```ruby
+2.3.3 :010 > 3.14.length
+NoMethodError: undefined method 'length' for 3.14:Float
+```
+
+Just as fish don't have a "knit" method, `Float`s don't know how long they are.
+
+Attributes are tiny little facts that an instance knows about itself that it
+**_returns_** when asked.
+
+## Define "Dot-notation" Syntax
+
+We access an attribute by adding a `.` and then the `attribute name` after the
+instance or a variable pointing to the instance. We discovered `"hi"`'s class by
+asking it for its `.class` attribute.
+
+## Define Method
+
+Ruby instances have "internal" methods. You didn't define them by writing
+`def...end` somewhere. The makers of Ruby did! While attributes report simple
+facts, methods are activities that instances know how to do. In Ruby, a
+`String` knows how to **_return_** a `reverse`d or `capitalize`d version of
+itself.
+
+```ruby
+2.3.3 :011 > mirror = "olleh"
+#=> "olleh"
+2.3.3 :012 > mirror.reverse
+#=> "hello"
+2.3.3 :014 > mirror.capitalize
+#=> "Olleh"
+```
+
+Just as before, an `Integer` does not know how to `.reverse`:
+
+```ruby
+2.3.3 :013 > 2112.reverse
+NoMethodError: undefined method 'reverse' for 2112:Fixnum
+```
+
+**_What methods or attributes an instance has available depends on its class._**
+
+## Demonstrate Method Chaining
+
+Remember, expressions in Ruby always have a return value. And we just
+learned that methods and attributes _return_ things. This means we can chain
+"dot-notation" calls to transform data.
+
+```ruby
+2.3.3 :011 > mirror = "olleh"
+#=> "olleh"
+2.3.3 :017 > mirror.reverse.capitalize
+#=> "Hello"
+```
+
+As a sneak peek of what this means, we will be able to do things like:
+
+```ruby
+first_place = sales_team.sort.first
+puts "#{first_place.name} gets coffee; coffee is for closers"
+```
+
+Method chaining allows us to express powerful ideas with very little typing. In
+some ways, this idea is now surpassing conversation: it's allowing us to
+express complex human ideas in tiny words separated by `.`. Amazing!
+
+## Find Documentation About Methods and Attributes
+
+We know you're really excited about `reverse` and `capitalize`, but are there
+other attributes and methods supported by `String`? Of course! We can look in
+the Ruby documentation for [`String`][string]. Or you could look at the
+information for `Fixnum` (which is confusingly documented under
```

Comment:
I would say take this out. In their minds, Ruby has Integers and Floats and Fixnum never existed since they'll probably never experience Ruby < 2.4